js/interface.js

```diff
@@ -82,11 +99,51 @@ function renderDataSources(dataSources) {
   dataSources.forEach(function (dataSource) {
     html.push(getDataSourceRender(dataSource));
   });
+
   $dataSources.html(html.join(''));
   $initialSpinnerLoading.removeClass('animated');
   $contents.removeClass('hidden');
 }
 
+function renderError(options) {
+  if (options === 'string') {
+    options = {
+      message: options
+    };
+  }
+
+  options = options || {};
+  options.message = options.message || 'Unexpected error';
+  var parsedError = Fliplet.parseError(options.error);
+
+  if (!parsedError) {
+    Fliplet.Modal.alert({
+      message: options.message
+    });
+    return;
+  }
+
+  Fliplet.Modal.confirm({
+    message: options.message,
+    buttons: {
+      cancel: {
```

Comment:
does pressing "Details" actually do anything?

Reply:
ah, nevermind. I've seen what you're done. It's a bit hard to read but works.

Reply:
Yes, it will show the parsed error message (if it exists).